src/optimagic/optimization/internal_optimization_problem.py

```diff
@@ -471,6 +475,8 @@ def _pure_evaluate_jac(
         out_jac = _process_jac_value(
             value=jac_value, direction=self._direction, converter=self._converter, x=x
         )
+        # Check for infinite values in the user-provided gradient
+        self._check_infinite_gradients(params, out_jac, jac_value)
```

Comment:
The comment is not necessary. Should be self-explanatory given the methods name. Also applies to the instances below.